src/app/profiles/profiles.editor.modal.ts

```diff
@@ -50,131 +48,101 @@ export class ProfilesEditorModalComponent {
     selector: any;
     public tinymce_plugin: TinyMCEPlugin;
     constructor(
+        public profileService: ProfileService,
         public rmm: RMM,
         private location: Location,
         public snackBar: MatSnackBar,
         public dialog_ref: MatDialogRef<ProfilesEditorModalComponent>,
         public draftDeskservice: DraftDeskService,
-        @Inject(MAT_DIALOG_DATA) public data: any
+        @Inject(MAT_DIALOG_DATA) public identity: Identity
     ) {
         this.tinymce_plugin = new TinyMCEPlugin();
-        if (data && data.type) {
-            this.type = data.type;
-            delete data.type;
-        }
-        if (data.profile && data.profile.email) {
-            this.set_localpart(data);
+        if (identity.email) {
```

Comment:
This comes before the `if(!identity)` check, will this cause an issue with `if (undefined.email)`?